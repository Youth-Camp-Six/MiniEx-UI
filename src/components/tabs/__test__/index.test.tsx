/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Tabs } from '..';

describe('Breadcrumb', () => {
  // 快照测试
  it('test all TabsProps ', () => {
    const options = [
      { label: '北京', value: '1' },
      { label: '上海', value: '2' },
      { label: '广州', value: '3' },
      { label: '深圳', value: '4' },
      { label: '乌鲁木齐', value: '5' },
    ];
    const wrapper = render(
      <div>
        <Tabs options={options}></Tabs> <br />
        <br />
        <Tabs
          round='round'
          options={[
            { label: 'Light', value: '1' },
            { label: 'Dark', value: '2' },
          ]}
        ></Tabs>
        <br />
        <br />
        <Tabs type='line' options={options}></Tabs>
        <br />
        <br />
        <Tabs itemWidth='120px' options={options}></Tabs>
      </div>
    );
    expect(wrapper.container.firstChild).toMatchSnapshot();
  });
});
