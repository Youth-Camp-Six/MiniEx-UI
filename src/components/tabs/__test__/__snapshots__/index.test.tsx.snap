// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb test all TabsProps  1`] = `
<div>
  <div
    class="mi-tabs mi-tabs-block"
  >
    <div
      class="mi-tabs-item-animation-active"
      style="transform: translateX(0px); width: 0px;"
    />
    <div
      class="mi-tabs-item mi-tabs-item-active"
    >
      北京
    </div>
    <div
      class="mi-tabs-item "
    >
      上海
    </div>
    <div
      class="mi-tabs-item "
    >
      广州
    </div>
    <div
      class="mi-tabs-item "
    >
      深圳
    </div>
    <div
      class="mi-tabs-item "
    >
      乌鲁木齐
    </div>
  </div>
   
  <br />
  <br />
  <div
    class="mi-tabs mi-tabs-block mi-tabs-round"
  >
    <div
      class="mi-tabs-item-animation-active"
      style="transform: translateX(0px); width: 0px;"
    />
    <div
      class="mi-tabs-item mi-tabs-item-active"
    >
      Light
    </div>
    <div
      class="mi-tabs-item "
    >
      Dark
    </div>
  </div>
  <br />
  <br />
  <div
    class="mi-tabs mi-tabs-line"
  >
    <div
      class="mi-tabs-item-animation-active"
      style="transform: translateX(0px); width: 0px;"
    />
    <div
      class="mi-tabs-item mi-tabs-item-active"
    >
      北京
    </div>
    <div
      class="mi-tabs-item "
    >
      上海
    </div>
    <div
      class="mi-tabs-item "
    >
      广州
    </div>
    <div
      class="mi-tabs-item "
    >
      深圳
    </div>
    <div
      class="mi-tabs-item "
    >
      乌鲁木齐
    </div>
  </div>
  <br />
  <br />
  <div
    class="mi-tabs mi-tabs-block"
  >
    <div
      class="mi-tabs-item-animation-active"
      style="transform: translateX(0px); width: 0px;"
    />
    <div
      class="mi-tabs-item mi-tabs-item-active"
      style="width: 120px;"
    >
      北京
    </div>
    <div
      class="mi-tabs-item "
      style="width: 120px;"
    >
      上海
    </div>
    <div
      class="mi-tabs-item "
      style="width: 120px;"
    >
      广州
    </div>
    <div
      class="mi-tabs-item "
      style="width: 120px;"
    >
      深圳
    </div>
    <div
      class="mi-tabs-item "
      style="width: 120px;"
    >
      乌鲁木齐
    </div>
  </div>
</div>
`;
